File: backend/POS.Core/Inventory/Item.cs
using System.ComponentModel.DataAnnotations;

namespace POS.Core.Inventory;

public sealed class Item : BaseEntity
{
    // TODO: set max length for all text fields
    [MaxLength(200)] public string Title { get; set; } = "";
    [MaxLength(100)] public string Serial { get; set; } = "";
    [MaxLength(200)] public string Description { get; set; } = "";
    [DataType("DECIMAL(19, 4)")] public decimal SalePrice { get; set; }
}

File: frontend/src/lib/data/Item.ts
export class Item {
    title: string;
    serial: string;
    description: string;
    salePrice: number;
    id?: number;
  
    constructor(
        title: string,
        serial: string,
        description: string,
        salePrice: number,
        id?: number
    ) {
      this.id = id;
      this.title = title;
      this.serial = serial;
      this.description = description;
      this.salePrice = salePrice;
    }
  }
  export function cloneItem(i): Item {
    return new Item(
      i.title,
      i.serial,
      i.description,
      i.salePrice,
      i.id,
    );
  }
  